NestJS/src/gmail_api/gmailapi.js

```diff
@@ -37,130 +29,148 @@ exports.getAllMessages = function(){
-    oAuth2Client.setCredentials(JSON.parse(token));
-
-    const a = await callback(oAuth2Client, data, recipient)
-    return a
 
+  const oAuth2Client = new google.auth.OAuth2(client_id, client_secret, redirect_uris);
+
+  const token = fs.readFileSync(TOKEN_PATH)
+  if (!token) return getNewToken(oAuth2Client, callback, data, recipient);
+  oAuth2Client.setCredentials(JSON.parse(token));
+
+  const result = await callback(oAuth2Client, data, recipient)
+  return result
 }
 
+
 /**
  * Get and store new token after prompting for user authorization, and then
  * execute the given callback with the authorized OAuth2 client.
  * @param {google.auth.OAuth2} oAuth2Client The OAuth2 client to get token for.
  * @param {getEventsCallback} callback The callback for the authorized client.
  */
-function getNewToken(oAuth2Client, callback, data, recipient) {
-  const authUrl = oAuth2Client.generateAuthUrl({
-    access_type: 'offline',
-    scope: SCOPES,
-  });
-  console.log('Authorize this app by visiting this url:', authUrl);
-  const rl = readline.createInterface({
-    input: process.stdin,
-    output: process.stdout,
-  });
-  rl.question('Enter the code from that page here: ', (code) => {
-    rl.close();
-    oAuth2Client.getToken(code, (err, token) => {
-      if (err) return callback(err);
-      oAuth2Client.setCredentials(token);
-      // Store the token to disk for later program executions
-      fs.writeFile(TOKEN_PATH, JSON.stringify(token), (err) => {
-        if (err) return console.error(err);
-        console.log('Token stored to', TOKEN_PATH);
-      });
-      callback(oAuth2Client, data, recipient);
-    });
-  });
-}
+
+ getNewToken = (oAuth2Client, callback, data, recipient) => {
+   const authUrl = oAuth2Client.generateAuthUrl({
+     access_type: 'offline',
+     scope: SCOPES,
+   });
+   console.log('Authorize this app by visiting this url:', authUrl);
+   const rl = readline.createInterface({
+     input: process.stdin,
+     output: process.stdout,
+   });
+   rl.question('Enter the code from that page here: ', (code) => {
+     rl.close();
+     oAuth2Client.getToken(code, (err, token) => {
+       if (err) return callback(err);
+       oAuth2Client.setCredentials(token);
+       // Store the token to disk for later program executions
+       fs.writeFile(TOKEN_PATH, JSON.stringify(token), (err) => {
+         if (err) return console.error(err);
+         console.log('Token stored to', TOKEN_PATH);
+       });
+       callback(oAuth2Client, data, recipient);
+     });
+   });
+ }
+
 
 /**
  * Lists the labels in the user's account.
  *
  * @param {google.auth.OAuth2} auth An authorized OAuth2 client.
  */
-function listLabels(auth) {
+
+
+getMessagesByDate = async (auth,date,callback) => {
+
   const gmail = google.gmail({version: 'v1', auth});
+  date = new Date(date).getTime() / 1000;
+  const query = 'is:inbox AND after: ${date}';
 
-  gmail.users.messages.list({
+  const getMail = await gmail.users.messages.list({
```

Comment:
variables should be a noun, functions - verb

Reply:
Ok, thanks